File: app/src/main/java/com/lumiyaviewer/lumiya/ui/settings/FriendlyEditTextPreference.java
package com.lumiyaviewer.lumiya.ui.settings;

import android.content.Context;
import androidx.preference.EditTextPreference;
import android.text.TextUtils;
import android.util.AttributeSet;

public class FriendlyEditTextPreference extends EditTextPreference {
    public FriendlyEditTextPreference(Context context) {
        super(context);
    }

    public FriendlyEditTextPreference(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
    }

    public FriendlyEditTextPreference(Context context, AttributeSet attributeSet, int i) {
        super(context, attributeSet, i);
    }

    @Override
    public CharSequence getSummary() {
        CharSequence summary;
        String text = getText();
        if (TextUtils.isEmpty(text) || (summary = super.getSummary()) == null) {
            return null;
        }
        return String.format(summary.toString(), text);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/ui/settings/FriendlyEditTextPreference.kt
package com.lumiyaviewer.lumiya.ui.settings

import android.content.Context
import androidx.preference.EditTextPreference
import android.text.TextUtils
import android.util.AttributeSet

class FriendlyEditTextPreference @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = androidx.preference.R.attr.editTextPreferenceStyle
) : EditTextPreference(context, attrs, defStyleAttr) {

    override fun getSummary(): CharSequence? {
        val text = text
        if (TextUtils.isEmpty(text)) {
            return null
        }
        val summary = super.getSummary() ?: return null
        return formatSummary(summary.toString(), text)
    }

    companion object {
        /**
         * Pure formatting rule: substitutes the preference's current text into
         * its summary template, matching `String.format(summary, text)`.
         */
        @JvmStatic
        fun formatSummary(summaryTemplate: String, text: String?): String =
            String.format(summaryTemplate, text)
    }
}
